go/libraries/doltcore/doltdb/doltdb.go

```diff
@@ -1042,6 +1040,58 @@ func (ddb *DoltDB) UpdateWorkingSet(
 	return err
 }
 
+// CommitWithWorkingSet combines the functionality of CommitWithParents with UpdateWorking set, and takes a combination
+// of their parameters. It's a way to update the working set and current HEAD in the same atomic transaction. It commits
+// to disk a pending commit value previously created with NewPendingCommit, asserting that the working set hash given
+// is still current for that HEAD.
+func (ddb *DoltDB) CommitWithWorkingSet(
+	ctx context.Context,
+	headRef ref.DoltRef, workingSetRef ref.WorkingSetRef,
+	commit *PendingCommit, workingSet *WorkingSet,
+	prevHash hash.Hash,
+	meta *WorkingSetMeta,
+) error {
+	wsDs, err := ddb.db.GetDataset(ctx, workingSetRef.String())
+	if err != nil {
+		return err
+	}
+
+	headDs, err := ddb.db.GetDataset(ctx, headRef.String())
+	if err != nil {
+		return err
+	}
+
+	// logrus.Tracef("Updating working set with root %s", workingSet.RootValue().DebugString(ctx, true))
+
+	workingRootRef, stagedRef, mergeStateRef, err := workingSet.writeValues(ctx, ddb)
+	if err != nil {
+		return err
+	}
+
+	// While we still have places that need user info threaded through, we're lenient on providing the meta
```

Comment:
Should we keep a counter of these things and log it or something? Or should we create a special value at the call sites similar to `context.TODO()` so that we can grep for it easily?